### PR TITLE
Add operandVersions to the ClusterOperator's Status

### PR DESCRIPTION
**Pull request by asalkeld**
Done as per request: https://github.com/openshift/cluster-baremetal-operator/pull/32#discussion_r500446673

### CODE REVIEW

File: controllers/clusteroperator.go [hunk 1]
```diff
@@ -58,6 +58,20 @@ func relatedObjects() []osconfigv1.ObjectReference {
 	}
 }
 
+// operandVersions returns the current list of OperandVersions for the
+// ClusterOperator objects's status.
+func (r *ProvisioningReconciler) operandVersions() []osconfigv1.OperandVersion {
+	if r.ReleaseVersion == "" {
```
Comment (kirankt) [inline controllers/clusteroperator.go:64]: Why can't we define an empty OperandVersion before we check for the env var. If the env var is defined we add values to the previously declared OperandVersion. That way, we only have to return once. Something like this perhaps?

```
        operandVersion := []osconfigv1.OperandVersion{}
        if r.releaseVersion != "" {
                operandVersions = append(operandVersions, osconfigv1.OperandVersion{Name: "operator", Version: r.releaseVersion})
        } 
        return operandVersion
```

**Review by kirankt — CHANGES_REQUESTED** [review]
Let a small nit. Please let me know what you think.

File: controllers/clusteroperator.go (hunk 1 above)
Comment (asalkeld) [reply controllers/clusteroperator.go:64]: sure, i can do that

**Pull request merged**